### PR TITLE
Projects can now contain services

### PR DESCRIPTION
**Pull request by jits**
- Only hub admins and project admins (aka managers) can create, edit and delete a project's services
- Only hub admins and project members can view services within a project

### CODE REVIEW

File: platform-hub-api/app/controllers/services_controller.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+class ServicesController < ApiJsonController
+
+  before_action :find_project
+  before_action :find_service, only: [ :show, :update, :destroy ]
+
+  # GET /projects/foo/services
+  def index
+    authorize! :read_services_in_project, @project
+
+    @services = @project.services.order(:name)
```
[the remark this answers is not in the document]
Comment (jits) [reply platform-hub-api/app/controllers/services_controller.rb]: Good catch

File: platform-hub-api/app/models/project.rb [hunk 2]
```diff
@@ -27,4 +27,7 @@ def should_generate_new_friendly_id?
     through: :memberships,
     source: :user
 
+  has_many :services,
+    dependent: :delete_all
```
[the remark this answers is not in the document]
Comment (jits) [reply platform-hub-api/app/models/project.rb]: Yeah, one of the design decisions I've noted down is we need to figure out what we do about projects and project services deletion, especially once we have other resources "attached" to them (like tokens, etc.) – soft delete / archived / disabled makes sense. We can address this once we get to integrating the tokens self serve (we'll probably remove the delete APIs and have something else instead. For now, at least in this PR, the effect of deleting stuff shouldn't cause any problems.

[the remark this answers is not in the document]
Comment (jits) [reply platform-hub-api/app/models/project.rb]: For your tokens refactor, we don't currently have a `DELETE` cluster API, right? In which case we can keep the cascading delete in the model, but it won't get used until someone manually deletes a cluster from the console (in which case we can probably live with tokens being deleted too).

File: platform-hub-api/db/migrate/20170920154859_create_services.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateServices < ActiveRecord::Migration[5.0]
+  def change
+    create_table :services, id: :uuid do |t|
+      t.string :name, null: false
+      t.text :description, null: false
+      t.belongs_to :project, type: :uuid
```
[the remark this answers is not in the document]
Comment (jits) [reply platform-hub-api/db/migrate/20170920154859_create_services.rb]: Yeah, good call.

File: platform-hub-api/app/models/project.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcinc) [reply platform-hub-api/app/models/project.rb]: 👍

**Pull request merged**